packages/eyeglass/src/importers/ModuleImporter.ts

```diff
@@ -1,57 +1,44 @@
-"use strict";
-
+import {readFileSync, existsSync} from "fs";
 import * as path from "path";
 import { NameExpander } from "../util/NameExpander";
-import * as fileUtils from "../util/files";
 import ImportUtilities from "./ImportUtilities";
+import { ImporterFactory, ImportedFile } from "./ImporterFactory";
+import { unreachable } from "../util/assertions";
+import { ImporterReturnType } from "node-sass";
+import { isPresent } from "../util/typescriptUtils";
+
+type ImportResultCallback =
+  (err: Error | null, data?: ImportedFile) => void;
 
 /*
- * Asynchronously walks the file list until a match is found. If
+ * Walks the file list until a match is found. If
  * no matches are found, calls the callback with an error
  */
-function readFirstFile(uri, possibleFiles, callback) {
-  // we'll keep track of the files we've tried for error handling
-  var triedFiles = [];
-
-  // iterates through the possibleFiles until one is successfully read
-  // errors if none are readable
-  function tryNextFile() {
-    // get the next file
-    var nextFile = possibleFiles.next().value;
-    // if there is a next file...
-    if (nextFile) {
-      // keep track of it
-      triedFiles.push(nextFile);
-      // and try to read it
-      fileUtils.readFile(nextFile, "utf8", function(err, data) {
-        // if it failed, keep trying
-        if (err) {
-          tryNextFile();
-        } else {
-          // if it didn't fail, we found the first file so return it
-          callback(null, {
-            contents: data.toString(),
-            file: nextFile
-          });
-        }
+function readFirstFile(uri: string, possibleFiles: Set<string>, callback: ImportResultCallback) {
+  for (let nextFile of possibleFiles) {
+    try {
+      let data = readFileSync(nextFile, "utf8");
+      // if it didn't fail, we found the first file so return it
+      callback(null, {
+        contents: data.toString(),
+        file: nextFile
       });
-    } else {
-      // could not read any of the possible files, so raise an error
-      var errorMsg = [
-        "Could not import " + uri + " from any of the following locations:"
-      ].concat(triedFiles).join("\n  ");
-      callback(new Error(errorMsg));
+      return;
+    } catch {
+      // pass
```

Comment:
I concur, but I don't think I introduced a new hurdle with this in this PR -- rather I only exposed this in a more visible way by changing this API to be properly synchronous instead of wrapped by an async API